<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CryptoAnalysisClass</class>
 <widget class="QMainWindow" name="CryptoAnalysisClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1174</width>
    <height>760</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CryptoAnalysis</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_7">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_13">
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <family>Times New Roman</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Alphabet: </string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLineEdit" name="alphabetLE">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="clearButtonEnabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="caseSensitiveCheckBox">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Case sensitive</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Plaintext</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="plaintextTE">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>10</pointsize>
               </font>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="analyzePlaintextPB">
                <property name="text">
                 <string>Analyze</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_14">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <item>
                   <widget class="QLabel" name="aSBLabel">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="text">
                     <string>A:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="aSB">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="bSBLabel">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="text">
                     <string>B:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="bSB">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QPushButton" name="encryptPB">
                  <property name="text">
                   <string>Encrypt</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Ciphertext</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="ciphertextTE">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>10</pointsize>
               </font>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="analyzeCiphertextPB">
                <property name="text">
                 <string>Analyze</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="decryptPB">
                <property name="text">
                 <string>Auto-decrypt</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTabWidget" name="statsTW">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
         </font>
        </property>
        <property name="currentIndex">
         <number>2</number>
        </property>
        <widget class="QWidget" name="statsCharsTab">
         <attribute name="title">
          <string>Chars</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QTabWidget" name="charsTW">
            <property name="currentIndex">
             <number>2</number>
            </property>
            <widget class="QWidget" name="charsChartLexTab">
             <attribute name="title">
              <string>Chart Alphabetical</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QChartView" name="charsLexChartView">
                <property name="font">
                 <font>
                  <pointsize>12</pointsize>
                 </font>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="charsChartFreqTab">
             <attribute name="title">
              <string>Chart by Frequency (Desc)</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QChartView" name="charsFreqChartView">
                <property name="font">
                 <font>
                  <pointsize>12</pointsize>
                 </font>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="charsTableTab">
             <attribute name="title">
              <string>Table</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QTableWidget" name="charsTableWidget">
                <property name="alternatingRowColors">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="statsBigramsTab">
         <attribute name="title">
          <string>Bigrams</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QTabWidget" name="bigramsTW">
            <property name="currentIndex">
             <number>1</number>
            </property>
            <widget class="QWidget" name="bigramsChartTab">
             <attribute name="title">
              <string>Chart</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QChartView" name="bigramsChartView">
                <property name="font">
                 <font>
                  <pointsize>12</pointsize>
                 </font>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="bigramsTableTab">
             <attribute name="title">
              <string>Table</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QTableWidget" name="bigramsTableWidget">
                <property name="alternatingRowColors">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QScrollArea" name="bigramsMatrixTab">
             <attribute name="title">
              <string>Matrix</string>
             </attribute>
             <widget class="MatrixColorPlot" name="bigramsMCP">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>696</width>
                <height>211</height>
               </rect>
              </property>
             </widget>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="statsTrigramsTab">
         <attribute name="title">
          <string>Trigrams</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QTabWidget" name="trigramsTW">
            <property name="currentIndex">
             <number>1</number>
            </property>
            <widget class="QWidget" name="trigramsChartTab">
             <attribute name="title">
              <string>Chart</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QChartView" name="trigramsChartView">
                <property name="font">
                 <font>
                  <pointsize>12</pointsize>
                 </font>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="trigramsTableTab">
             <attribute name="title">
              <string>Table</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QTableWidget" name="trigramsTableWidget">
                <property name="alternatingRowColors">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1174</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenPlaintext"/>
    <addaction name="actionSavePlaintext"/>
    <addaction name="actionOpenCiphertext"/>
    <addaction name="actionSaveCiphertext"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpenPlaintext">
   <property name="text">
    <string>Open Plaintext</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionSavePlaintext">
   <property name="text">
    <string>Save Plaintext</string>
   </property>
  </action>
  <action name="actionOpenCiphertext">
   <property name="text">
    <string>Open Ciphertext</string>
   </property>
  </action>
  <action name="actionSaveCiphertext">
   <property name="text">
    <string>Save Ciphertext</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
  <customwidget>
   <class>MatrixColorPlot</class>
   <extends>QWidget</extends>
   <header>MatrixColorPlot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="CryptoAnalysis.qrc"/>
 </resources>
 <connections/>
</ui>
